src/v/archival/ntp_archiver_service.cc

```diff
@@ -552,11 +593,13 @@ ss::future<ntp_archiver::scheduled_upload> ntp_archiver::schedule_single_upload(
               }
               return rtx;
           });
+    auto is_compacted = upload.source->is_compacted_segment()
+                        && upload.source->finished_self_compaction();
     co_return scheduled_upload{
       .result = std::move(upl_fut),
       .inclusive_last_offset = offset,
       .meta = cloud_storage::partition_manifest::segment_meta{
```

Comment:
here you need to create a v2 segment_meta

Reply:
I assume it is just an extra version field? I have added now.